internal/dag/builder.go

```diff
@@ -590,6 +597,9 @@ func (b *Builder) computeRoutes(sw *ObjectStatusWriter, proxy *projcontour.HTTPP
 		}
 		routes = append(routes, r)
 	}
+	// Mark proxy as valid
+	sw.SetValid()
```

Comment:
I don't think this is the right place for this. I believe the call should happen in the caller of this method.

Reply:
I'll look again to see if I can do that, but what I found was since it's called recursively, the children proxies don't get status set, only the final one. 

This is also breaking my other status check for the same reason, so will work to flush it out.